beacon-chain/sync/subscriber_test.go

```diff
@@ -92,6 +92,7 @@ func TestSubscribe_ReceivesAttesterSlashing(t *testing.T) {
 	topic := "/eth2/%x/attester_slashing"
 	var wg sync.WaitGroup
 	wg.Add(1)
+	params.SetupTestConfigCleanup(t)
```

Comment:
I think this is still going to be fragile. How will we ensure users always clean up the config?

Reply:
Whenever using a resource, you either created it or get reference for it. If you create a resource (say database to be used in test), then you need to tear it down. In #5725 I made sure that when db resource is created, teardown is automatically attached. There weren't any checks to ensure users clean up - it was done manually.

In case of configuration which is global variable to which we want to keep accessing by pointer, then there's no other option but to register cleanup function (and you need to register it **before** you modify pointed data). The `SetupTestConfigCleanup()` is exactly that - registering teardown function. At the moment we do nothing to teardown - we just modify the configuration object, with `SetupTestConfigCleanup()` we have a way to attach cleanup routine - all the rest of the code remains the same.

To sum up: when setting up resource, teardown is responsibility of users, especially when it comes to *writing tests*. We do not ensure it in any way (and I am not sure how we are supposed to do so, except for some static analysis tool). We are using global var, pointer to that global var, and if we continue to do so - then I am not sure what else can be done.

Currently, we are leaking the state - and we cannot avoid it. The `SetupTestConfigCleanup()` function allows you to restore the state.